interactions/models/internal/listener.py

```diff
@@ -73,6 +77,14 @@ def wrapper(coro: AsyncCallable) -> "Listener":
             if not asyncio.iscoroutinefunction(coro):
                 raise TypeError("Listener must be a coroutine")
 
+            coro_parametrers = inspect.signature(coro).parameters.copy()
+            if coro_parametrers and list(coro_parametrers.keys())[0] == "self":
+                # remove self; those who name it differently are on their own
```

Comment:
I see how it is 😂

Reply:
Unfortunately as far as I was able to find it's impossible to tell if the decorated function is a method othervise. `inspect.ismethod` won't work because the decorated function becomes method only *after* being decorated.

Reply:
I think I have a workaround. Just got home. I'll have a toy in a bit

Reply:
8d3141cce765e496c98cfddf3593d61b25723307 should resolve this for all parties